### PR TITLE
[rust] add parser

### PR DESCRIPTION
**Pull request by prateekmedia**
**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.
- [ ] **I have mentioned this change in the [changelog](https://github.com/CCExtractor/ccextractor/blob/master/docs/CHANGES.TXT).**

**My familiarity with the project is as follows (check one):**

- [ ] I have never used CCExtractor.
- [ ] I have used CCExtractor just a couple of times.
- [ ] I absolutely love CCExtractor, but have not contributed previously.
- [x] I am an active contributor to CCExtractor.

---

Rewrote argument parsing with all the structs and enums too into rust.

Help arguments demo:
image

Closed old PR as sample platform broke for it: #1547

### CODE REVIEW

File: .github/workflows/build_mac.yml [hunk 1]
```diff
@@ -74,22 +74,27 @@ jobs:
...
+  #   - name: cmake
+  #     run: |
+  #       mkdir build && cd build
+  #       cmake -DWITH_OCR=ON -DWITH_HARDSUBX=ON ../src
+  #   - name: build
+  #     run: |
+  #       make -j$(nproc)
+  #     working-directory: build
+  #   - name: Display version information
+  #     run: ./build/ccextractor --version
```
Comment (PunitLodha) [inline .github/workflows/build_mac.yml:97]: Shouldn't this be uncommented?

File: .github/workflows/test_rust.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            src/rust/.cargo/registry
+            src/rust/.cargo/git
+            src/rust/target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-cargo-
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+      - name: rustfmt
```
Comment (PunitLodha) [inline .github/workflows/test_rust.yml:35]: Should be named test

File: windows/rust.bat [hunk 3]
```diff
@@ -6,5 +6,5 @@ cd ..\..\windows
 IF "%~1"=="-r" (
 copy x86_64-pc-windows-msvc\release\ccx_rust.lib .\ccx_rust.lib
 ) ELSE (
-copy x86_64-pc-windows-msvc\debug\ccx_rust.lib .\ccx_rust.lib
+copy x86_64-pc-windows-msvc\release-with-debug\ccx_rust.lib .\ccx_rust.lib
```
Comment (PunitLodha) [inline windows/rust.bat:9]: Why do we have release with debug here?

File: src/rust/Cargo.toml [hunk 4]
```diff
@@ -15,15 +15,26 @@ log = "0.4.0"
...
 [build-dependencies]
 bindgen = "0.64.0"
+pkg-config = "0.3.30"
 
 [features]
 hardsubx_ocr = ["rsmpeg", "tesseract-sys", "leptonica-sys"]
+
+[profile.release-with-debug]
+inherits = "release"
+debug = true
```
Comment (PunitLodha) [inline src/rust/Cargo.toml:40]: Same here

**Review by PunitLodha — APPROVED** [review]

**Comment by PunitLodha** [comment]
@prateekmedia also fix the merge conflicts

**Comment by prateekmedia** [comment]
@PunitLodha Done!

**Comment by prateekmedia** [comment]
@PunitLodha Done!

**Review by PunitLodha — APPROVED** [review]

**Comment by IshanGrover2004** [comment]
@prateekmedia @PunitLodha  Are the sample test ran on this PR, Because I tested some samples on current CCExtractor code & its crashes which was working fine before this merging happens (I checked on each commit through when it is failing, found out this is the one which causes).

**Comment by prateekmedia** [comment]
@IshanGrover2004 The teletext PR will resolve the crashes currently.

**Pull request merged**